system/jlib/jtrace.cpp

```diff
@@ -107,6 +108,293 @@ class NoopSpanExporterFactory
     }
 };
 
+/**
+ * Converts an OpenTelemetry span kind to its string representation.
+ * These are OTel span kinds defined in api/include/opentelemetry/trace/span.h,
+ * not HPCC JLib CSpan kinds
+ *
+ * @param spanKind The OpenTelemetry span kind to convert.
+ * @return The string representation of the OpenTelemetry span kind.
+ */
+const char * spanKindToString(opentelemetry::trace::SpanKind spanKind)
+{
+    switch (spanKind)
+    {
+        case opentelemetry::trace::SpanKind::kClient:
+            return "Client";
+        case opentelemetry::trace::SpanKind::kServer:
+            return "Server";
+        case opentelemetry::trace::SpanKind::kProducer:
+            return "Producer";
+        case opentelemetry::trace::SpanKind::kConsumer:
+            return "Consumer";
+        case opentelemetry::trace::SpanKind::kInternal:
+            return "Internal";
+        default:
+            return "Unknown";
+    }
+}
+
+enum JLogExportLevel
+{
+    JLogExportLevelNone = 0,
+    JLogExportLevelMin = 1,
+    JLogExportLevelDefault = 2,
+    JLogExportLevelMax = 3
+};
+
+class JLogSpanExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+private:
+    JLogExportLevel logLevel = JLogExportLevelDefault;
+public:
+
+    inline JLogExportLevel jlogLevelFromStr(char const * str)
+    {
+        if (!isEmptyString(str))
+        {
+            if(strnicmp(str, "NONE", 4)==0)
+                return JLogExportLevelNone;
+            if(strnicmp(str, "MIN", 3)==0)
+                return JLogExportLevelMin;
+            if(strnicmp(str, "DEFAULT", 7)==0)
+                return JLogExportLevelDefault;
+            if(strnicmp(str, "MAX", 3)==0)
+                return JLogExportLevelMax;
+        }
+
+        return JLogExportLevelDefault;
+    }
+    JLogSpanExporter() {}
+
+    JLogSpanExporter(const char * logLevel)
+    {
+        setLogLevel(jlogLevelFromStr(logLevel));
+    }
+    
+    JLogSpanExporter(JLogExportLevel logLevel)
+    {
+        setLogLevel(logLevel);
+    }
+
+    void setLogLevel(JLogExportLevel level)
+    {
+        logLevel = level;
+    }
+
+    /**
+    * @return Returns a unique pointer to an empty recordable object
+    */
+    std::unique_ptr<opentelemetry::sdk::trace::Recordable> MakeRecordable() noexcept override
```

Comment:
clearer to the reader if this has virtual on the front (even though not required by the standard)